### PR TITLE
feat: boilerplate to support development version api v5 pt.1 (WPB-1124)

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

This PR is part one of finalizing api v4, contains 

- Creating boilerplate code to support v5 endpoints

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases and review with:

- https://github.com/wireapp/kalium/pull/2011

- [x] GitHub link to other pull request

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by MohamadJaara** [comment]
the PR name says that it will move v4 to supported but 
1. i do not see this in the change 
2. can we hold on on this, api v4 is not released so we first need to have the backend team move to v4 and only then we can have the switch on our app, just to not have the server suddenly release v4 with a bug

**Comment by yamilmedina** [comment]
> the PR name says that it will move v4 to supported but
> 
> 1. i do not see this in the change
> 2. can we hold on on this, api v4 is not released so we first need to have the backend team move to v4 and only then we can have the switch on our app, just to not have the server suddenly release v4 with a bug

I converted it to a draft, it is a WIP, as I need to open one extra PR 😅

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v5/authenticated/PropertiesApiV5.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.kalium.network.api.v5.authenticated
+
+import com.wire.kalium.network.AuthenticatedNetworkClient
+import com.wire.kalium.network.api.v4.authenticated.PropertiesApiV4
+
+internal open class PropertiesApiV5 internal constructor(
+    private val authenticatedNetworkClient: AuthenticatedNetworkClient,
```
Comment (mchenani) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v5/authenticated/PropertiesApiV5.kt:25]:
```suggestion
    private val authenticatedNetworkClient: AuthenticatedNetworkClient
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v5/unauthenticated/networkContainer/UnauthenticatedNetworkContainerV5.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.wire.kalium.network.networkContainer.UnauthenticatedNetworkClientProviderImpl
+import com.wire.kalium.network.networkContainer.UnauthenticatedNetworkContainer
+import com.wire.kalium.network.tools.ServerConfigDTO
+import io.ktor.client.engine.HttpClientEngine
+
+class UnauthenticatedNetworkContainerV5 internal constructor(
+    networkStateObserver: NetworkStateObserver,
+    backendLinks: ServerConfigDTO,
+    proxyCredentials: ProxyCredentialsDTO?,
+    engine: HttpClientEngine = defaultHttpEngine(backendLinks.links.apiProxy, proxyCredentials),
```
Comment (mchenani) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v5/unauthenticated/networkContainer/UnauthenticatedNetworkContainerV5.kt:46]:
```suggestion
    engine: HttpClientEngine = defaultHttpEngine(backendLinks.links.apiProxy, proxyCredentials)
```

**Review by mchenani — APPROVED** [review]
🔥, just two tiny suggestions.

**Review by typfel — APPROVED** [review]

**Pull request merged**